src/OpenTelemetry/Trace/BatchExportActivityProcessorOptions.cs

```diff
@@ -78,13 +83,7 @@ private static bool TryLoadEnvVarInt(string envVarKey, out int result)
                 return false;
             }
 
-            if (!int.TryParse(value, out var parsedValue))
-            {
-                OpenTelemetrySdkEventSource.Log.FailedToParseEnvironmentVariable(envVarKey, value);
-                return false;
-            }
-
-            result = parsedValue;
+            result = int.Parse(value); // throws FormatException or OverflowException for bad input
```

Comment:
I understand the desire to avoid a new exception, however, the root cause won't be clear since it doesn't include the `envVarKey` and its `value`. A wrapper exception that makes clear the root cause seems desirable (perhaps it can be generic enough so it can be re-used in other locations).

Reply:
I agree that finding a way to include both the `envVarKey` and its `value`. Could we format a message and `throw` some existing somewhat appropriate exception like an `ArgumentException`?

Reply:
_Waiting until we decide if we want to fail-fast or just log the error and continue with defaults_. [More](https://github.com/open-telemetry/opentelemetry-dotnet/pull/2395#issuecomment-924119879)